src/core/io.c

```diff
@@ -127,35 +129,80 @@ static Janet cfun_io_popen(int32_t argc, Janet *argv) {
 #define popen _popen
 #endif
     FILE *f = popen((const char *)fname, (const char *)fmode);
-    if (!f) {
+    
+    if(NULL == f) {
         return janet_wrap_nil();
+    } else {
```

Comment:
I think instead of copy pasting the same code 3 times, a static utility could be shared.

Reply:
or part of the logic could be in makef